wagtail/models/__init__.py

```diff
@@ -1299,6 +1299,143 @@ def __init__(self, *args, **kwargs):
     def __str__(self):
         return self.title
 
+    def _process_leaf(self, nodes):
+        """
+        Process children to add to a leaf node
+        """
+        # Get number of children
+
+        step = 1
+        max_length = self.__class__._meta.get_field("path").max_length
+        for node in nodes:
+            node.depth = self.depth + 1
+            node.path = self.__class__._get_path(self.path, node.depth, step)
+            step += 1
+            if len(node.path) > max_length:
+                raise ValidationError(
+                    "The new node is too deep in the tree, try \
+                      increasing the path.max_length property \
+                    and UPDATE your database"
+                )
+            self.numchild += 1
+            node._cached_parent_obj = self
+
+        return
+
+    def _process_unordered_children(self, nodes):
+        """
+        Create child nodes without any specific order.
+        """
+        last_child = self.get_last_child()
+        max_length = self.__class__._meta.get_field("path").max_length
+        for node in nodes:
+            node.depth = self.depth + 1
+            node.path = last_child._inc_path()
+            last_child = node
+            if len(node.path) > max_length:
+                raise ValidationError(
+                    "The new node is too deep in the tree, try \
+                      increasing the path.max_length property \
+                    and UPDATE your database"
+                )
+            self.numchild += 1
+            node._cached_parent_obj = self
+
+        return
+
+    def _process_child_nodes(self, nodes):
+        """
+        Process the child nodes of this page, setting their paths and depths
+        as appropriate and also updating the parent's numchild. This is performed by the treebeard library when saving
+        the page to the database using their inbuilt tree functionality.
+        """
+
+        if self.is_leaf():
+            return self._process_leaf(nodes)
```

Comment:
Naming thing is hard, and a lot of these methods could benefit from a bit of a rethink. In the context of these additions, they might seem fine, but when you consider the codebase on the whole, names benefit from more detail to distinguish them from existing (or future) similarly named methods.

See how you go with moving the functionality to the Manager first of all, then we can look at the naming in future.

Tip for future: Discussions about naming can often derail an otherwise non-controversial PR, demotivate everybody involved in the process, and reducing the likelihood of changes being merged. You can side-step such problems early on by keeping your additions together in a single method, and using things like comments to explain better what is happening (it also makes code easier to review). It can always be split out at the end if deemed necessary!

Reply:
Thanks for the tip! I'll keep that in mind 🙌 

I didn't know how naming methods would impact the codebase so I didn't think as well as I should've before adding it. Sorry about that.

Reply:
No problem! learning is all part of the process :)